### PR TITLE
fix #282165: Mute voice produces note that doesn't stop when overlap present

### PR DESCRIPTION
**Pull request by mattmcclinch**
Resolves: https://musescore.org/en/node/282165.

This uses the same logic as in `Seq::playEvent()` for determining whether or not the event corresponds to a muted track.

### CODE REVIEW

**Comment by anatoly-os** [comment]
@mattmcclinch thanks! Will a MIDI test catch the bug the fix? If so, could you please add one?

**Comment by mattmcclinch** [comment]
I added a test which uncovered a bug in MIDI export. I have fixed this bug as well, using the same logic as before.

File: synthesizer/event.cpp [hunk 1]
```diff
@@ -391,8 +394,18 @@ void EventMap::fixupMIDI()
 
       auto it = begin();
       while (it != end()) {
+            NPlayEvent event = it->second;
```
Comment (anatoly-os) [inline synthesizer/event.cpp]: Isn't it possible to avoid code duplication? Extract a method?

Comment (mattmcclinch) [reply synthesizer/event.cpp]: Yes, I agree that it is better to avoid code duplication. I just pushed a change to this branch that does what you suggest.

**Pull request merged**